models/user.go

```diff
@@ -503,6 +503,23 @@ func (u *User) GetRepositoryIDs(units ...UnitType) ([]int64, error) {
 	return ids, sess.Where("owner_id = ?", u.ID).Find(&ids)
 }
 
+// GetActiveRepositoryIDs returns non-archived repositories IDs where user owned and has unittypes
+// Caller shall check that units is not globally disabled
+func (u *User) GetActiveRepositoryIDs(units ...UnitType) ([]int64, error) {
```

Comment:
This will have duplicated ids

Reply:
If more than one `UnitType` is passed to the func, you mean? True.

Since xorm doesn't seem to have a `FindDistinct()` func I could use instead of 

```go
return ids, sess.Where("owner_id = ?", u.ID).Find(&ids)
```
... the only thing I can think of is explicitly clearing duplicates from `ids` before I return them. Somehow I think there must be a better way. Any suggestions on how to handle that more elegantly?

Reply:
Or you can use `GroupBy` like you did in line 555

Reply:
Done, thanks!